### DISCUSSION TITLE
Synchronous Serial Rx

### DISCUSSION THREAD

**Discussion by Crsarmv7l** [jgromes/RadioLib#1016] [Q&A] [answered]
Hi,

I bumped a relevant old thread but this probably deserves its own. 

Some background:
Both this lib and LSatan's ELECHOUSE lib heavily utilize the FIFO buffer for user ease of use. Unfortunately I can't use the FIFO as it requires a preamble/syncword (high/low), and I am looking to turn all filtering for RX off except Carrier Sense. That leave Sync/Async Serial RX.

I have been studying the Ti Application Note AN095 - Continuous Data Streaming Applications, and it seems like Sync Serial would provide the easiest method as this is a hobbyist project.
 
It seems like for Sync Serial RX I would simply have to read the Clock pin, if the Clock pin is high (assuming inverse register isn't set), read the Data pin for the bit.

This is appealing to me as the Clock is high in accordance with the datarate. I know that this setup will be a bit error prone given the lack of oversampling which is acceptable to me.

I have 2 questions:

1. Am I missing anything in my understanding/idea?
2. I see in the RadioLib api directReceive which I assume is Sync Serial, as there is directAsyncReceive?

**Comment by jgromes** [comment]
Check out this example: https://github.com/jgromes/RadioLib/blob/master/examples/SX127x/SX127x_Receive_Direct/SX127x_Receive_Direct.ino

It does what you want on SX127x, however, RadioLib keeps (mostly) the same interface for all radio modules. So you should be able to use that as a starting point.

> I see in the RadioLib api directReceive which I assume is Sync Serial, as there is directAsyncReceive?

Both are available, `CC1101::receiveDirect()` and `CC1101::receiveDirectAsync()`.

**Comment by Crsarmv7l** [comment]
I'll look at that example.

But to make sure, directReceive is sync serial?

**Reply by jgromes** [reply, marked as answer]
Yes it is - I will amend the doxygen comments to state that explicitly.

**Comment by Crsarmv7l** [comment]
Awesome, thank you!

I have already written my code using ELECHOUSE so I'll do a bit more
testing with that, but if it doesn't easily fall into place, I'm switch to
radiolib. It's maintained, and you have been very helpful.
